### PR TITLE
Admin tap: supports multiple attached requests

### PR DESCRIPTION
**Pull request by xu1zhou**
Commit Message: 
Additional Description: Currently, admin tap support only single attached request to capture traffic for observation, this feature allow user to attach more request to admin tap until reach the num in `max_concurrent_streams`  added in AdminConfig API. 
Risk Level: Low
Testing: Added unit tests for multiple attached request.
Docs Changes:  tap filter doc example add `max_concurrent_streams` filed
Release Notes: N/A

### CODE REVIEW

File: api/envoy/extensions/common/tap/v3/common.proto [hunk 1]
```diff
@@ -42,4 +42,5 @@ message AdminConfig {
   // Opaque configuration ID. When requests are made to the admin handler, the passed opaque ID is
   // matched to the configured filter opaque ID to determine which filter to configure.
   string config_id = 1 [(validate.rules).string = {min_len: 1}];
+  uint64 max_concurrent_streams = 2 [(validate.rules).uint64 = {gt: 0}];
```
Comment (abeyad) [inline api/envoy/extensions/common/tap/v3/common.proto:45]: Other `max_concurrent_streams` values in the APIs use `google.protobuf.UInt32Value`, e.g. https://github.com/envoyproxy/envoy/blob/d82b5a02befbff3b2119b309e865f7b44d207907/api/envoy/config/core/v3/protocol.proto#L63.

So I suggest we use the same.

Also, can we add a comment for what this new field means?  And is there a default value if it's not specified?

Also, I think `gte: 1` makes it clearer than `gt: 0` (it took me a couple seconds to realize the value must be >= 1), but that's more of a personal preference.  The couple other instances of `max_concurrent_streams` in proto definitions also use `gte: 1`

Comment (xu1zhou) [reply api/envoy/extensions/common/tap/v3/common.proto:45]: That make sense, I'll add some comments for this new field.

**Comment by yanavlasov** [comment]
@xu1zhou the `//test/extensions/filters/http/tap:tap_filter_integration_test` is now failing with SIGSEGV https://source.cloud.google.com/results/invocations/b26cf29d-c4f4-4226-978f-3c7b0bbc6228

**Pull request closed without merging**